File: labs/lab9/q3.py
#Code by Mudit
#Date: 23/09/2026
from itertools import groupby  # Groups consecutive equal elements

k = int(input("Enter k: "))
a = list(map(int, input("Enter the entries: ").split()))

answer = 0
position = 0

for value, group in groupby(a):
    length = len(list(group))  # Finds length of the consecutive group

    if value == 1 and length > k:
        answer = position + k + 1  # Finds first session exceeding k
        break

    position += length  # Moves to the next group

print("Output =", answer)

File: labs/lab9/q3.c
//Code by Mudit
//Date: 23/09/2026
#include <stdio.h>

// Finds consecutive 1s starting from position i
int runLength(int a[], int n, int i)
{
    if (a[i] == 0)
        return 0;

    int count = 0;

    while (i < n && a[i] == 1)
    {
        count++;
        i++;
    }

    return count;
}

int main()
{
    int n, k;

    printf("Enter n: ");
    scanf("%d", &n);

    printf("Enter k: ");
    scanf("%d", &k);

    int a[n];

    printf("Enter the entries: ");

    for (int i = 0; i < n; i++)
        scanf("%d", &a[i]);

    int answer = 0;

    // Check each position for a violation
    for (int i = 0; i < n; i++)
    {
        int length = runLength(a, n, i);

        if (length > k)
        {
            // Convert 0-based index to session number
            answer = i + k + 1;
            break;
        }
    }

    printf("Output = %d\n", answer);

    return 0;
}
